### PR TITLE
[FORMS-9320] accordion component implementation

### PR DESCRIPTION
**Pull request by Aabha1**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/material-ui-based-components/src/components/Accordion.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import AccordionSummary from '@mui/material/AccordionSummary';
+import AccordionDetails from '@mui/material/AccordionDetails';
+import Typography from '@mui/material/Typography';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+
+const AccordionComponent = function (props: PROPS_PANEL) {
+    const [expanded, setExpanded] = React.useState<number>(0);
+    // @ts-ignore
+    const mappings = useContext(FormContext).mappings;
+    const items = props.items || [];
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/Accordion.tsx:16]: Hidden items should not render here. Please filter props.item with the visible property.

File: packages/material-ui-based-components/src/components/Accordion.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    const handleChange = (index: number) => {
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/Accordion.tsx:18]: bind with useCallback hook

File: packages/material-ui-based-components/src/components/Accordion.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        setExpanded(index);
+    };
+
+    const getChild = (child: any, index: number) => {
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/Accordion.tsx:22]: bind with useCallback hook

File: packages/material-ui-based-components/src/components/Accordion.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        const Comp = getRenderer(child, mappings);
+        return Comp ? <Comp key={`${child.id}_${index}`} {...child} /> : (null);
+    };
+
+
+    return (
+        <div>
+            <div>{props?.label?.value ? <div>{props.label.value}</div> : null}</div>
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/Accordion.tsx:30]: add this condition on div

File: packages/material-ui-based-components/src/components/Accordion.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            {
+                items.map((child: any, index: number) => (
+                    <Accordion expanded={expanded === index} onChange={() => { handleChange(index); }} key={`${child.id}_${index}`}>
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/Accordion.tsx:33]: value of the key can not be the same(line 24 and 33)

File: packages/material-ui-based-components/src/components/Accordion.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        const Comp = getRenderer(child, mappings);
+        return Comp ? <Comp key={`${child.id}_${index}`} {...child} /> : (null);
+    };
+
+
+    return (
+        <div>
+            <div>{props?.label?.value ? <div>{props.label.value}</div> : null}</div>
+            {
+                items.map((child: any, index: number) => (
```
Comment (amit-mnnit) [inline packages/material-ui-based-components/src/components/Accordion.tsx:32]: Please add disabled check

File: packages/material-ui-based-components/src/components/Accordion.tsx (hunk 1 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/Accordion.tsx:16]: Checked as items= props.visible?props.items:[];

File: packages/material-ui-based-components/src/components/Accordion.tsx (hunk 2 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/Accordion.tsx:18]: Binded with useCallback

File: packages/material-ui-based-components/src/components/Accordion.tsx (hunk 4 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/Accordion.tsx:30]: Added to div

File: packages/material-ui-based-components/src/components/Accordion.tsx (hunk 5 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/Accordion.tsx:33]: Changed value of one of the keys

File: packages/material-ui-based-components/src/components/Accordion.tsx (hunk 6 above)
Comment (Aabha1) [reply packages/material-ui-based-components/src/components/Accordion.tsx:32]: Added disabled check

**Pull request merged**